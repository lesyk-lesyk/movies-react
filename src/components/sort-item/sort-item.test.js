import React from 'react';
import { SortItem } from './index';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store'

describe('SortItem', () => {
  it('renders with active class', () => {
    const store = configureStore()({ searchForm: { sortBy: 'SMTH' } });
    const callback = () => { };
    const tree = renderer
      .create(<SortItem sortBy='SMTH' onClick={callback} store={store} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without active class', () => {
    const store = configureStore()({ searchForm: { sortBy: 'SMTH' } });
    const callback = () => { };
    const tree = renderer
      .create(<SortItem sortBy='OTHER' onClick={callback} store={store} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should run callback on click', () => {
    const store = configureStore()({ searchForm: { sortBy: 'RATING' } });
    const callback = jest.fn();

    const wrapper = shallow(<SortItem sortBy='RATING' onClick={callback} store={store} />);
    wrapper.first().dive().find('.sort-item').simulate('click', { preventDefault() { } });

    expect(callback).toHaveBeenCalled();
  });
});
