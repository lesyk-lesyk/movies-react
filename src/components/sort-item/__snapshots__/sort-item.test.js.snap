// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortItem renders with active class 1`] = `
<span
  className="sort-item active"
  onClick={[Function]}
/>
`;

exports[`SortItem renders without active class 1`] = `
<span
  className="sort-item"
  onClick={[Function]}
/>
`;
